consensus/istanbul/validator/weighted.go

```diff
@@ -359,6 +377,19 @@ func (valSet *weightedCouncil) SubListWithProposer(prevHash common.Hash, propose
 		return []istanbul.Validator{proposer}
 	}
 
+	rules := fork.Rules(new(big.Int).SetUint64(valSet.blockNum + 1))
+	// After Randao: SelectRandaoCommittee
+	if rules.IsRandao {
+		// This committee must include proposers for all rounds because
+		// the proposer is picked from the this committee. See weightedRandomProposer().
+		if valSet.mixHash == nil {
+			logger.Error("no mixHash", "number", valSet.blockNum)
+			return nil
+		}
+		return SelectRandaoCommittee(validators, committeeSize, valSet.mixHash)
```

Comment:
Don't we need the following nil check?
```suggestion
		if committee := SelectRandaoCommittee(validators, committeeSize, valSet.mixHash); committee != nil {
		    return committee
		}
		return validator 
```